lib/Agrammon/UI/CommandLine.pm6

```diff
@@ -95,6 +97,44 @@ multi sub MAIN('run', Str $filename, ExistingFileOrStdin $input, ExistingFile :$
     say $output;
 }
 
+#| Get inputs from dataset
+multi sub MAIN(
+        'getDataset', Str $model-filename, Str $username, Str $dataset-name,
+        Str $version, Str $gui, Str $variant, ExistingFile :$cfg-file
+    ) is export {
+    my %data := get-dataset $model-filename, $username, $dataset-name, $version, $gui, $variant, :$cfg-file;
+    my @output;
+    @output.push("## User: %data<username>");
+    @output.push("## Dataset: %data<dataset>");
+    if %data<validation-errors> {
+        note "WARNING: validation errors";
+        @output.push('## Validation errors');
+        for @(%data<validation-errors>) -> $error {
+            @output.push($error.message);
+        }
+        @output.push('##');
+    }
+    my $inputs = %data<inputs>;
+    for $inputs.kv -> $module, $_ {
+        when Hash {
+            for sorted-kv($_) -> $input, $value {
+                @output.push("$module;$input;$value");
+            }
+        }
+        when Array {
+            for sorted-kv($_) -> $instance-id, %instance-inputs {
+                for sorted-kv(%instance-inputs) -> $fq-name, %values {
+                    my $q-name = $module ~ '[' ~ $instance-id ~ ']' ~ $fq-name.substr($module.chars);
```

Comment:
I suspect we do this calculation (of the name from module, instance ID, and fully qualified name) in enough places while producing output that it may just be worth factoring out into ` Agrammon::OutputFormatter::Util`.

Reply:
Done